metal/constants.go

```diff
@@ -0,0 +1,16 @@
+package metal
+
+const (
+	configMapResource         = "configmaps"
+	hostnameKey               = "kubernetes.io/hostname"
+	emIdentifier              = "equinixcloud-provider-equinix-metal-auto"
```

Comment:
There are a few references in this PR to `equinixcloud-provider-equinix-metal-*`.
Are these arbitrary names? Can we change them to `cloud-provider-equinix-metal-*` without impact?

Reply:
It is somewhat inherited, but those can and have changed. We can change that without much issue.